exoscale/resource_exoscale_nlb.go

```diff
@@ -0,0 +1,275 @@
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutRead))
+	defer cancel()
+
+	nlb, err := findNlb(ctx, d, meta)
+	if err != nil {
+		return false, err
+	}
+
+	if nlb == nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func resourceNlbUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: beginning update", resourceNlbIDString(d))
+
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutUpdate))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	nlb, err := findNlb(ctx, d, meta)
+	if err != nil {
+		return err
+	}
+
+	if nlb == nil {
+		return fmt.Errorf("Network Load Balancer %q not found", d.Id())
+	}
+
+	d.Partial(true)
+
+	if d.HasChange("name") {
+		nlb.Name = d.Get("name").(string)
+		d.SetPartial("name")
+	}
+
+	if d.HasChange("description") {
+		nlb.Description = d.Get("description").(string)
+		d.SetPartial("description")
+	}
+
+	_, err = client.UpdateNetworkLoadBalancer(
+		ctx,
+		d.Get("zone").(string),
+		nlb,
+	)
+
+	if err != nil {
+		return err
+	}
+
+	d.Partial(false)
+
+	log.Printf("[DEBUG] %s: update finished successfully", resourceNlbIDString(d))
+
+	return resourceNlbRead(d, meta)
+}
+
+func resourceNlbDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: beginning delete", resourceNlbIDString(d))
+
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutDelete))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	err := client.DeleteNetworkLoadBalancer(
+		ctx,
+		d.Get("zone").(string),
+		d.Id(),
+	)
+
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: delete finished successfully", resourceNlbIDString(d))
+
+	return nil
+}
+
+func resourceNlbApply(ctx context.Context, client *egoscale.Client, d *schema.ResourceData, nlb *egoscale.NetworkLoadBalancer) error {
+	if err := d.Set("name", nlb.Name); err != nil {
+		return err
+	}
+
+	if err := d.Set("description", nlb.Description); err != nil {
+		return err
+	}
+
+	if err := d.Set("created_at", nlb.CreatedAt.String()); err != nil {
+		return err
+	}
+
+	if err := d.Set("state", nlb.State); err != nil {
+		return err
+	}
+
+	if err := d.Set("ip_address", nlb.IPAddress.String()); err != nil {
+		return err
+	}
+
+	services := make([]string, len(nlb.Services))
+	for i, service := range nlb.Services {
+		services[i] = service.Name
+	}
+
+	if err := d.Set("services", services); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func findNlb(ctx context.Context, d *schema.ResourceData, meta interface{}) (*egoscale.NetworkLoadBalancer, error) {
+	client := GetComputeClient(meta)
+
+	resp, err := client.RequestWithContext(ctx, egoscale.ListZones{})
+	if err != nil {
+		return nil, err
+	}
+	zones := resp.(*egoscale.ListZonesResponse).Zone
+
+	var nlb *egoscale.NetworkLoadBalancer
+	for _, zone := range zones {
+		n, err := client.GetNetworkLoadBalancer(
+			ctx,
+			zone.Name,
+			d.Id(),
+		)
```

Comment:
Let's go for the compact format here, since parameters are short it's perfectly readable 😉

```suggestion
		n, err := client.GetNetworkLoadBalancer(ctx, zone.Name, d.Id())
```

Reply:
ccb328f2e6db70d8e23c79b0b2de24f71f472e01